### ISSUE TITLE
Split out announcement config, enable date-bounding

### ISSUE THREAD

**Issue by liggitt** [kubernetes/website#25759]
**This is a Feature Request**

**What would you like to be added**

* Separate the config driving the display and content of the announcement banner on the website to its own dir, with steering approvers
* Add the ability to specify a start and end date for an announcement, and adjust the display logic to only render the announcement between those dates, so no PR is required to enable or disable the announcement

**Why is this needed**

To help implement the policy described in https://github.com/kubernetes/community/blob/master/communication/website-guidelines.md

**Comments**

One possible approach would be to 

1. load the announcement config from a dedicated JSON file (e.g. `announcements/config.json`) using [getJSON](https://gohugo.io/templates/data-templates/#data-driven-content):
    ```json
    {"items":[
      {
        "begin":       "2020-11-23",
        "end":         "2020-12-14",
        "background":  "#000000",
        "_":           "// # See /i18n/en.toml for message text and title.",
        "title_key":   "announcement_title",
        "message_key": "announcement_message"
      },
      {
        "begin":       "2020-12-20",
        "end":         "2021-01-01",
        "background":  "#228B22",
        "_":           "// # See /i18n/en.toml for message text and title.",
        "title_key":   "announcement_holidays_title",
        "message_key": "announcement_holidays_message"
      }
    ]}
    ```

2. infer whether an announcement is active based on whether there is an item with an active date range

3. adjust the announcement partial to use the first announcement with an active date range

cc @sftim

**Comment by liggitt** [comment]
cc @kubernetes/steering-committee

**Comment by sftim** [comment]
I'd like to try implementing this. I favor YAML over JSON as YAML supports multiline strings natively.

**Comment by sftim** [comment]
I think this will be 2 PRs:
1. Add a GitHub Action to trigger automatic site deploys
1. Set the announcement banner based on a Hugo data source.

**Comment by sftim** [comment]
Alternative: use JavaScript client side to select the banner display logic. That'd slow down rendering for _every_ page load so I don't recommend it.

**Comment by liggitt** [comment]
+1 for avoiding dynamic client-side javascript load

I messed around with something inside the data folder and hugo was very unhappy with an OWNERS file

**Comment by sftim** [comment]
Hmm. I could use a GitHub `CODEOWNERS` file. Alternatively, the data could be fetched via HTTP. Neither is ideal.

**Comment by sftim** [comment]
We _could_ use `CODEOWNERS` for this specific though - does that feel like a good idea? It'd mean that an announcement change needs approval from `OWNERS` (to keep Prow happy) and `CODEOWNERS` (to keep GitHub happy).

**Comment by kbhawkey** [comment]
I left comments in PR #25769 before reading this issue.
Since the announcement messages require approval by the Steering committee to publish, it may make sense to create the announcement data files in  a different repository (separate owners) and share with the site for publishing. Docs would maintain the templates to publish the announcement to the site (and maintain a base or default announcement data file) and handle the publishing processes.

**Comment by sftim** [comment]
BTW, Netlify doesn't leave a Hugo process to make live reloading work. Live reloading is there for local development.

I'm wary of the HTTP fetch; at the moment, it introduces a hard dependency for both local development (every page relies on fetching that JSON) and for site publishes. I'd sooner use `CODEOWNERS` or implement the approval requirements with an extra people step.
AIUI, there's no demand for a high frequency or low latency for announcement updates, so an extra step might be OK.

@liggitt / steering committee, any thoughts on those requirements?

**Comment by liggitt** [comment]
> I messed around with something inside the data folder and hugo was very unhappy with an OWNERS file

ha, I take it back, we just have to tweak our ignored file regexp, and then an OWNERS file in the data dir seemed to work fine:

```diff
diff --git a/config.toml b/config.toml
index ac1c1845f..33309b8aa 100644
--- a/config.toml
+++ b/config.toml
@@ -13,7 +13,7 @@ disableBrowserError = true
 
 disableKinds = ["taxonomy", "taxonomyTerm"]
 
-ignoreFiles = [ "^OWNERS$", "README[-]+[a-z]*\\.md", "^node_modules$", "content/en/docs/doc-contributor-tools" ]
+ignoreFiles = [ "(^|/)OWNERS$", "README[-]+[a-z]*\\.md", "^node_modules$", "content/en/docs/doc-contributor-tools" ]
 
 timeout = 3000
```

**Comment by liggitt** [comment]
I think keeping all the content here, and putting an OWNERS file in the announcements data dir would be simpler. I'd also like to avoid the http fetch.

Something like this could work:
```diff
diff --git a/OWNERS_ALIASES b/OWNERS_ALIASES
index 3d1ae50af..605b5c95b 100644
--- a/OWNERS_ALIASES
+++ b/OWNERS_ALIASES
@@ -215,3 +215,11 @@ aliases:
     - idvoretskyi
     - MaxymVlasov
     - Potapy4
+  steering-committee: # PR approvals for announcements dir
+    - cblecker
+    - derekwaynecarr
+    - dims
+    - liggitt
+    - mrbobbytables
+    - nikhita
+    - parispittman

diff --git a/data/announcements/OWNERS b/data/announcements/OWNERS
new file mode 100644
index 000000000..887ed974d
--- /dev/null
+++ b/data/announcements/OWNERS
@@ -0,0 +1,8 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+# Disable inheritance as content in this dir is scoped to steering approval per 
+# https://github.com/kubernetes/community/blob/master/communication/website-guidelines.md
+options:
+  no_parent_owners: true
+approvers:
+- steering-committee # defined in OWNERS_ALIASES
```

**Comment by sftim** [comment]
> https://github.com/kubernetes/website/issues/25759#issuecomment-750389599

Sounds good to me. I'll add that.

**Comment by liggitt** [comment]
Resolved by #25771 and #25769. Thanks!
